actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -833,6 +833,27 @@ def fetch(key, *args)
       )
     end
 
+    # Returns parameters for the given keys. If a key can't be found, there are
+    # several options: With no other arguments, it will raise an
+    # ActionController::ParameterMissing error; if a block is given, then that will
+    # be run and its result returned for the missing key.
+    #
+    #     params = ActionController::Parameters.new(name: "Francesco", age: 22)
+    #     params.fetch_values(:name, :age)                # => ["Francesco", 22]
+    #     params.fetch_values(:name, :none)               # => ActionController::ParameterMissing: param is missing or the value is empty or invalid: none
+    #     params.fetch_values(:name, :none) { |key| key } # => ["Francesco", :none]
+    def fetch_values(*keys)
+      values = @parameters.fetch_values(*keys) do |missing_key|
+        missing_key = missing_key.to_sym if missing_key.respond_to?(:to_sym) && !keys.include?(missing_key)
+        if block_given?
+          yield missing_key
+        else
+          raise ActionController::ParameterMissing.new(missing_key, @parameters.keys)
+        end
+      end
+      values.map { |value| convert_value_to_parameters(value) }
```

Comment:
Would it be more efficient to invoke [Array#map!](https://docs.ruby-lang.org/en/master/Array.html#method-i-map-21), since the `values` array was allocated and populated on the prior line:

```suggestion
      values.map! { |value| convert_value_to_parameters(value) }
```

Reply:
Sure, manipulating in place would avoid allocating another array